crates/kornia-core-ops/src/ops.rs

```diff
@@ -0,0 +1,111 @@
+use kornia_core::{storage::TensorStorage, SafeTensorType, Tensor, TensorAllocator};
+
+use crate::error::TensorOpsError;
+
+/// Compute the sum of the elements in the tensor along dimension `dim`
+///
+pub fn sum_elements<T, const N: usize, A>(
```

Comment:
for future PR -- worth finding an abstraction like here https://github.com/kornia/kornia-rs/blob/main/crates/kornia-core/src/tensor.rs#L710

one more thought is how to support with the same interface tensors with contiguous memory that do not require computing the offset for the given index.

Reply:
and something I coudn't figure out yet is how to add a `keepdim` e.g to reduce and return the tensor rank to N -1. @emilmgeorge maybe you have some thoughts here

Reply:
@edgarriba 
I think we can simply remove the `dim` dimension from `shape` and `strides` when `keepdim == false` since the num elements stay the same. I hope I'm not misunderstanding what you meant.

Reply:
this is actually the only limitation we have with generic tensor ranks. Hopefully we con figure out a workaround as this comes useful to convert tensor to onnxruntime or any other dl engine :)

see: https://github.com/kornia/kornia-rs/blob/8cd3394ab7aad40f0d1ef503366eca7a98756b94/crates/kornia-dnn/src/rtdetr.rs#L158

Reply:
where BTW, we should be able to bindo directly to cuda at some point

Reply:
Would it be feasible to make dimensions dynamic? Like `Tensor<T, A>` instead of `Tensor<T, N, A>` and then manage shape/stride at runtime using Vec. I don't know the whole code base well enough yet to know the possibility and consequences of this. Just sharing a thought. If feasible, it would solve this problem and give more flexibility. But I guess this would also cause some performance hit.

Reply:
I tried to experiment with this idea of keeping const generic ranks and using it to inform the sum operation on what rank tensor to return. I've added these traits heavily borrowed (simplified and probably worsen) from [dfdx](https://github.com/coreylowman/dfdx/blob/main/dfdx-core/src/shapes/shape.rs) with this [commit](https://github.com/kornia/kornia-rs/pull/142/commits/4b36377d7e122d3a6a9beb79bb7431ef6e1ce79b). Example usage can be seen in the test from this [commit](https://github.com/kornia/kornia-rs/pull/142/commits/d4fa9512851aa36bb230e47d619e7c60faadeb3d).

I might have messed up this PR now. Let me know what's best to do next.

Reply:
@jandremarais that's a nice feature ! Maybe as a separated PR we can introduce the reshape stuff, and later integrate into the ops?